src/Microsoft.Health.Dicom.Core/Features/Operations/IDicomOperationsClient.cs

```diff
@@ -69,4 +69,19 @@ public interface IDicomOperationsClient
     /// <exception cref="ArgumentNullException"><paramref name="specification"/> is <see langword="null"/>.</exception>
     /// <exception cref="OperationCanceledException">The <paramref name="cancellationToken"/> was canceled.</exception>
     Task<Guid> StartExportingFilesAsync(ExportSpecification specification, CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously begins the instance blob migration
+    /// </summary>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The value of its <see cref="Task{TResult}.Result"/> property contains the ID of the operation
+    /// that is performing the asynchronous addition.</returns>
+    Task<Guid> StartBlobMigrationAsync(CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Asynchronously gets the completed status of blob migration
+    /// </summary>
+    /// <param name="cancellationToken">The token to monitor for cancellation requests. The default value is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>The value of its <see cref="Task{TResult}.Result"/> property contains the completed status</returns>
+    Task<bool> IsBlobMigrationCompletedAsync(CancellationToken cancellationToken = default);
```

Comment:
We would have the completion flag in table storage, so I suppose we have a service to get/set the flag, rather than through DicomOperationsClient.
Just food of thought:
```CSharp
interface IFlagService
{
    Task<string> GetFlagAsync(string name, CancellationToken cancellationToken = default)
    Task SetFlagAsync(string name, string value, CancellationToken cancellationToken = default)
}

```

Reply:
Talked with Will, check with the orchestration status is much easier, as avoid overhead of creating additional table. so I think this implementation is right, we just need to filter out Copy orchestration to be cleanedup. https://github.com/microsoft/healthcare-shared-components/blob/430626d31705323531b003f1796a175c1b5643e9/src/Microsoft.Health.Operations.Functions/Management/PurgeOrchestrationInstanceHistory.cs#L55

Reply:
Keeping the IsBlobCompleted method